rust_extension/src/test_fixtures/explicit_traceback.py

```
@@ -0,0 +1,27 @@
+"""Raise a nested exception and expose its parts with ``__traceback__`` cleared.
+
+Loaded by ``traceback_capture_tests`` via ``include_str!``. Running this module
+leaves ``exc_type``, ``exc_value`` and ``exc_tb`` in the supplied globals, with
+``exc_value.__traceback__`` set to ``None`` so that tests can check that a
+traceback passed explicitly in a ``(type, value, traceback)`` tuple is still
+used for frame extraction.
+"""
+
+
+def inner():
+    raise ValueError("test error")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)